src/Chirp.Core/ILikeRepository.cs

```diff
@@ -0,0 +1,12 @@
+namespace Chirp.Core;
+
+public interface ILikeRepository
+{
+    public void LikeCheep(int authorId, int cheepId);
+    public void UnlikeCheep(int authorId, int cheepId);
+    public bool LikeExists(int authorId, int cheepId);
+    public IEnumerable<Like> FindLikesByCheepId(int cheepId);
+    public IEnumerable<Like> FindLikesByAuthorId(int authorId);
+    public int FindLikeCountByCheepId(int cheepId);
+    public bool LikesOwnCheep(int authorId, int cheepId);
+}
```

Comment:
Here we are repeating code. Perhaps we should consider an interface for theses. 
Perhaps `IUserAction`?
```cs
public interface ILinkAction<T>
{
    public void PerformAction(int id1, int id2);
    public void UndoAction(int id1, int id2);
    public bool LinkExists(int id1, int id2);
    public IEnumerable<T> FindLinkById1(int id); // Shitty name
    public IEnumerable<T> FindLinkById2(int id); // Shitty name
}
```
The counts could be omitted since we can just use
```cs
 public IEnumerable<T> FindLinkById1(int id); // Shitty name
 public IEnumerable<T> FindLinkById2(int id); // Shitty name
```

Reply:
Let's implement this in a future pr